src/highrl/utils/teacher_utils.py

```diff
@@ -88,27 +87,13 @@ def get_obstacles_from_action(
     Returns:
         List[SingleObstalce: list of obtained obstacles
     """
-    x_func, y_func, w_func, h_func = action[4], action[5], action[6], action[7]
-    x_max = opt.width**cfg.max_obstacles_count
-    y_max = opt.height**cfg.max_obstacles_count
-
-    x_func = math.ceil(x_func * x_max)
-    w_func = math.ceil(w_func * x_max)
-    y_func = math.ceil(y_func * y_max)
-    h_func = math.ceil(h_func * y_max)
-
-    obstacles = []
-    for _ in range(cfg.max_obstacles_count):
-        obs_x = x_func % opt.width
-        obs_y = y_func % opt.height
-        obs_w = min(w_func % opt.width, opt.width - obs_x)
-        obs_h = min(h_func % opt.height, opt.height - obs_y)
-        obstacles.append(SingleObstacle(obs_x, obs_y, obs_w, obs_h))
-
-        x_func = x_func // opt.width
-        w_func = w_func // opt.width
-        y_func = y_func // opt.height
-        h_func = h_func // opt.height
+    obstacles_ls = []
+
+    # Convert obstacles position/dimension from [0, 1] to [0, width]
+    for idx in range(4, 40, 4):
+        dims = [action[idx + dim_i] * opt.width for dim_i in range(4)]
+        obstacles_ls.append(SingleObstacle(*dims))
+    obstacles = Obstacles(obstacles_ls)
```

Comment:
FATAL :x: :-1: 
The obstacles are assumed to come in different sizes. So please consider that.

Reply:
Why do you see this as constant size?
Are all actions with the same value?